### PR TITLE
[SDAO-90] Switch build setup to stack and haskell.nix

### PR DESCRIPTION
**Pull request by zhenyavinogradov**
[//]: # (This is a template of a pull request template.)
[//]: # (You should modify it considering specifics of a particular repository and put it there.)
[//]: # (Comments like this are meta-comments, they shouldn't be present in the final template.)
[//]: # (Comments starting with '

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [ ] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

[//]: # (Mostly for public repositories)
[//]: # (Recording changes is optional, depends on repository, useful for some libs)
- Public contracts
  - [ ] Any modifications of public contracts comply with the [Evolution
  of Public Contracts](https://www.notion.so/serokell/Evolution-of-Public-Contracts-2a3bf7971abe4806a24f63c84e7076c5) policy.
  - [ ] I added an entry to the [changelog](../tree/master/CHANGES.md) if my changes are visible to the users
        and
  - [ ] provided a migration guide for breaking changes if possible

#### Stylistic guide (mandatory)

[//]: # (Update link to style guide if necesary or remove if it's not present)

- [ ] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [ ] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: stkr-token/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    src = projectSrc;
+    filter = path: type:
+      let
+        relPath = pkgs.lib.removePrefix (toString projectSrc.origSrc + "/") (toString path);
+        regexes = [ "hpack" "hpack/.*" "stkr-token" "stkr-token/.*" ];
+      in pkgs.lib.any (re: builtins.match re relPath != null) regexes;
+  };
+
+  # haskell.nix does not support package.yaml files referencing
+  # parent directectories, so we generate the .cabal file ourselves
```
Comment (balsoft) [inline stkr-token/default.nix]: Is there an issue for this upstream?

**Comment by zhenyavinogradov** [comment]
Yeah that'll take a while

File: stkr-token/default.nix (hunk 1 above)
Comment (zhenyavinogradov) [reply stkr-token/default.nix]: I haven't found one, I think haskell.nix support for hpack is just quite limited.

Comment (balsoft) [reply stkr-token/default.nix]: Maybe create it?

Comment (zhenyavinogradov) [reply stkr-token/default.nix]: Created one

**Comment by zhenyavinogradov** [comment]
Rebased against master

**Pull request merged**